designer/client/context/__tests__/FeatureFlagContext.jest.tsx

```diff
@@ -0,0 +1,145 @@
+
+    const { findAllByText } = render(
+      <WrappingComponent>
+        <FeatureToggle feature="featureEditPageDuplicateButton">
+          <button>Johnny Five Is Alive!</button>
+        </FeatureToggle>
+      </WrappingComponent>
+    );
+    expect(await findAllByText("Johnny Five Is Alive!")).toBeTruthy();
+  });
+
+  test("should not show element if feature is not set", async () => {
+    const stub = sinon
+      .stub(FeatureToggleApi.prototype, "fetch")
+      .callsFake(async function () {
+        return { featureEditPageDuplicateButton: false };
+      });
+
+    const { queryAllByText } = render(
+      <WrappingComponent>
+        <FeatureToggle feature="featureEditPageDuplicateButton">
+          <button>Johnny Five Is Alive!</button>
+        </FeatureToggle>
+      </WrappingComponent>
+    );
+    await act(() => stub.getCall(0).returnValue);
+    expect(await queryAllByText("Johnny Five Is Alive!")).toHaveLength(0);
+  });
+
+  test("should not show element if feature is not defined", async () => {
+    const stub = sinon
+      .stub(FeatureToggleApi.prototype, "fetch")
+      .callsFake(async function () {
+        return { featureA: false };
+      });
+
+    const { queryAllByText } = render(
+      <WrappingComponent>
+        <FeatureToggle feature="featureB">
+          <button>Johnny Five Is Alive!</button>
+        </FeatureToggle>
+      </WrappingComponent>
+    );
+    await act(() => stub.getCall(0).returnValue);
+
+    expect(await queryAllByText("Johnny Five Is Alive!")).toHaveLength(0);
+  });
+
+  test("should feature toggle api only load once", async () => {
+    const stub = sinon
+      .stub(FeatureToggleApi.prototype, "fetch")
+      .callsFake(async function () {
+        return { featureA: false, featureB: true, featureC: true };
+      });
+
+    render(
+      <WrappingComponent>
+        <FeatureToggle feature="featureA">
+          <button>Johnny Five Is Alive!</button>
+        </FeatureToggle>
+        <FeatureToggle feature="featureB">
+          <button>Johnny Five Is Alive!</button>
+        </FeatureToggle>
+        <FeatureToggle feature="featureC">
+          <button>Johnny Five Is Alive!</button>
+        </FeatureToggle>
+      </WrappingComponent>
+    );
+    await act(() => stub.getCall(0).returnValue);
+
+    expect(stub.calledOnce).toBeTruthy();
+  });
+
+  test("should not show element if features api fails", async () => {
+    sinon.stub(FeatureToggleApi.prototype, "fetch").callsFake(function () {
+      return new Promise((_resolve, reject) => {
+        reject();
+      });
+    });
+
+    const { queryAllByText } = render(
+      <WrappingComponent>
+        <FeatureToggle feature="featureEditPageDuplicateButton">
+          <button>Johnny Five Is Alive!</button>
+        </FeatureToggle>
+      </WrappingComponent>
+    );
+    expect(await queryAllByText("Johnny Five Is Alive!")).toHaveLength(0);
+  });
+
+  it("should feature hook return feature context value", () => {
```

Comment:
maybe I need a coffee - I can't read this sentence

Reply:
useFeatures returns feature context value(??)

Reply:
you can see my delirium at the time from this name.

Reply:
moving to hook tests, although we dont really have hook tests

Reply:
that's ok - testing hooks is anti react-testing-library philosophy. with RTL mr. Dodds prefers we test the component using the hook, and checking the component renders as intended

Reply:
That seems like a very opinionated view on the "smallest possible unit" 😂  I have moved it, shall I move it back or delete the test?

Reply:
nope, not that important 😂